test/apps/iodemo/io_demo.cc

```diff
@@ -1225,6 +1237,7 @@ class DemoServer : public P2pDemoCommon {
                                     const UcxAmDesc &data_desc) {
         VERBOSE_LOG << "receiving AM IO write data";
         assert(msg->data_size != 0);
+        assert(msg->conn_id == conn->remote_id());
```

Comment:
you just set it in `handle_io_msg(msg, conn);` - it will never fail

Reply:
removed, it was leftover